cmd/bbolt/main.go

```diff
@@ -509,16 +502,16 @@ func (cmd *pageItemCommand) Run(args ...string) error {
 	return nil
 }
 
-// leafPageElement retrieves a leaf page element.
-func (cmd *pageItemCommand) leafPageElement(pageBytes []byte, index uint16) (*guts_cli.LeafPageElement, error) {
-	p := (*guts_cli.Page)(unsafe.Pointer(&pageBytes[0]))
+func (cmd *pageItemCommand) validateLeafPage(pageBytes []byte, index uint16) (*common.Page, error) {
```

Comment:
The previous `(*pageItemCommand) leafPageElement` actually includes two operations:
1. [validation](https://github.com/etcd-io/bbolt/blob/1273ac779e759591a3424a2bb225a0295eb0610c/cmd/bbolt/main.go#L514-L520)
2. [access content](https://github.com/etcd-io/bbolt/blob/1273ac779e759591a3424a2bb225a0295eb0610c/cmd/bbolt/main.go#L521)

I just extract the first operation into an explict method, namely `(*pageItemCommand) validateLeafPage`.

The main motivation is to make sure all `LeafPageElement[s]` / `BranchPageElement[s]` have consistent meaning. It's a little confusing that some of them means just accessing content, but others means validating & accessing content.

Reply:
I don't consider these validations... 
To me it's only a sanity check that confirms whether the method was used in a correct context.
We cannot access leaf elements on a page that is not a 'leaf' and we should go above the number of items on the page. 
I would just add similar 'assertions / sanity checks' on other accessors to be on the 'defensive' side.

Reply:
assertion seems not good for CMD. Usually we need to return an error (display an error message) instead of just panicking for CMD coming from users.

Let me revert this change if you don't have objections.

Reply:
I recall the other motivation of extracting the validation into an explicit method....  previously [(*pageItemCommand) leafPageElement](https://github.com/etcd-io/bbolt/blob/1273ac779e759591a3424a2bb225a0295eb0610c/cmd/bbolt/main.go#L513) returns `*guts_cli.LeafPageElement`, the struct `guts_cli.LeafPageElement` was exposed directly. In this PR, we only expose the method [(*Page) LeafPageElement()](https://github.com/ahrtr/bbolt/blob/0d34b53317ca5c25da3e34594a3bb93edd915465/internal/common/page.go#L76) instead of the struct [leafPageElement](https://github.com/ahrtr/bbolt/blob/0d34b53317ca5c25da3e34594a3bb93edd915465/internal/common/page.go#L232), so we can't return `common.leafPageElement` any more.

Anyway, updated. Please take a look at the last commit https://github.com/etcd-io/bbolt/pull/407/commits/f68adfee81834c5008e828b0bd418942400cc651